### PR TITLE
Improve PPO readability

### PR DESCRIPTION
**Pull request by alan-cooney**
Adds mostly comments and renames a few variables to be more descriptive. Also deletes two unused methods.

I've run an example with wandb to check this doesn't change anything unexpectedly - https://wandb.ai/alancooney/trlx/runs/aew4nu69 (same as before the changes)

Co-authored-by: @jezgillen

### CODE REVIEW

**Comment by alan-cooney** [comment]
@reciprocated to help with documenting this code, are you able to confirm what `for _ in range(self.n_updates_per_batch):` is intended to do in the base trainer?

https://github.com/CarperAI/trlx/blob/84a07113d8ced1497c5a7941a02f9c534bb4e185/trlx/trainer/accelerate_base_trainer.py#L445-L449

Git blame suggests this is your code. My understanding is it runs the same batch multiple times - is that as expected? That is to say it lets the user repeat a batch, rather than e.g. having multiple steps e.g. turns in a text-based game (which I don't think the library supports).

**Comment by reciprocated** [comment]
Hey there! `n_updates_per_batch` let's perform multiple gradient updates on the same batch of data, this is a common PPO implementation detail. Here is an excerpt from PPO's paper abstract [1] as reference:
> Whereas standard policy gradient methods perform one gradient update per data sample, we propose a novel objective function that enables multiple epochs of minibatch updates

[1] https://arxiv.org/abs/1707.06347

**Comment by alan-cooney** [comment]
> > Whereas standard policy gradient methods perform one gradient update per data sample, we propose a novel objective function that enables multiple epochs of minibatch updates

Awesome thanks for the quick reply! I'll add this in to the comments

File: trlx/orchestrator/ppo_orchestrator.py [hunk 1]
```diff
@@ -92,7 +114,7 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
                 )
                 for output in outputs
             ]
-            response_tensors = torch.vstack(outputs).to(device)
+            padded_samples = torch.vstack(outputs).to(device)
```
Comment (reciprocated) [inline trlx/orchestrator/ppo_orchestrator.py:117]: Probably `padded_outputs` (or rather `output_tensors` in view of `query_tensors`) is better suited name here, since by a sample `prompt + output` is implied everywhere else

File: trlx/orchestrator/ppo_orchestrator.py [hunk 2]
```diff
@@ -178,74 +200,98 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
...
-                logprobs = logprobs_from_logits(logits, all_tokens)
-                ref_logprobs = logprobs_from_logits(ref_logits, all_tokens)
+                logprobs = logprobs_of_labels(logits, all_tokens)
+                ref_logprobs = logprobs_of_labels(ref_logits, all_tokens)
 
-            n = samples.shape[0]
+            n_samples: int = samples.shape[0]
             logprobs = logprobs.cpu()
             ref_logprobs = ref_logprobs.cpu()
             query_tensors = query_tensors.cpu()
```
Comment (reciprocated) [inline trlx/orchestrator/ppo_orchestrator.py:214]: Might also rename `query` to `prompt` for consistency

File: trlx/trainer/accelerate_ppo_trainer.py [hunk 3]
```diff
@@ -17,37 +18,70 @@
...
 
+        # Setup rollout logging
         if config.train.rollout_logging_dir is not None:
             self.log_rollouts = True
             self.setup_rollout_logging(config)
         else:
             self.log_rollouts = False
 
+        # Setup the rollout store
+        # Rollouts contain the query & response, log probs, values
```
Comment (reciprocated) [inline trlx/trainer/accelerate_ppo_trainer.py:56]: prompt & output here for consistency again

**Review by reciprocated — CHANGES_REQUESTED** [review]
Splendid, what's left is to merge changes from the main and do a pre-commit run and I've also left a small suggestion to remain consistent in naming throughout. Thanks for the effort @alan-cooney @jezgillen

**Comment by LouisCastricato** [comment]
Let's get this merged

File: trlx/orchestrator/ppo_orchestrator.py (hunk 1 above)
Comment (alan-cooney) [reply trlx/orchestrator/ppo_orchestrator.py:117]: I think `padded_outputs` makes sense - made the change

File: trlx/orchestrator/ppo_orchestrator.py (hunk 2 above)
Comment (alan-cooney) [reply trlx/orchestrator/ppo_orchestrator.py:214]: Done

File: trlx/orchestrator/ppo_orchestrator.py (hunk 1 above)
Comment (alan-cooney) [reply trlx/orchestrator/ppo_orchestrator.py:117]: How about `sample_outputs` to be even clearer?

**Comment by alan-cooney** [comment]
Thanks for the review @reciprocated 

Note I've had to resolve a tonne of merge conflicts, so it's worth checking those are in line with expectations (they're primarily with your commits). 

Given the size of this PR please can you take a look as soon as possible & merge, otherwise we'll likely get other merge conflicts as well.

**Review by reciprocated — APPROVED** [review]
Marvellous, merging!

https://wandb.ai/sorry/trlx/reports/Improve-PPO-readability-210---VmlldzozNDg3ODEx

**Comment by alan-cooney** [comment]
Thanks for the quick review!
